### PR TITLE
[WT-40] Run 1st test - Basic UI Test at Login

### PR DESCRIPTION
**Pull request by steven-the-qa**
- Playwright project setup
- 1st UI test - Login page header text & route validation

### CODE REVIEW

File: playwright.config.ts [hunk 1]
```diff
@@ -24,88 +24,56 @@ const config: PlaywrightTestConfig = {
...
-    //   name: 'Google Chrome',
-    //   use: {
-    //     channel: 'chrome',
-    //   },
-    // },
   ],
 
   /* Folder for test artifacts such as screenshots, videos, traces, etc. */
-  // outputDir: 'test-results/',
+  outputDir: './test-results',
```
Comment (steven-the-qa) [inline playwright.config.ts]: This directory wasn't valid. Now the test-results folder has a report in it after tests finish.

File: playwright.config.ts [hunk 2]
```diff
@@ -24,88 +24,56 @@ const config: PlaywrightTestConfig = {
...
-     * Use the dev server by default for faster feedback loop.
-     * Use the preview server on CI for more realistic testing.
-    Playwright will re-use the local server if there is already a dev-server running.
-     */
-    command: process.env.CI ? 'vite preview --port 5173' : 'vite dev',
-    port: 5173,
-    reuseExistingServer: !process.env.CI
+    command: 'npm run dev',
+    url: 'http://localhost:5173',
+    timeout: 30000
```
Comment (steven-the-qa) [inline playwright.config.ts:88]: We need this configuration to run tests on a local server in CI

File: playwright.config.ts [hunk 3]
```diff
@@ -24,88 +24,56 @@ const config: PlaywrightTestConfig = {
...
-        ...devices['Desktop Firefox']
-      }
+        ...devices['Pixel 5'],
+      },
     },
-    {
-      name: 'webkit',
-      use: {
-        ...devices['Desktop Safari']
-      }
```
Comment (steven-the-qa) [inline playwright.config.ts:64]: I only want to support Mobile & Desktop Chrome for now. Chrome has the leading marketshare on both platforms.

File: playwright.config.ts [hunk 4]
```diff
@@ -24,88 +24,56 @@ const config: PlaywrightTestConfig = {
...
+    // PROD
+    // baseURL: 'https://dead-simple-weight-tracker.web.app',
 
     /* Collect trace when retrying the failed test. See https://playwright.dev/docs/trace-viewer */
     trace: 'on-first-retry',
-
-    /* Only on CI systems run the tests headless */
-    headless: !!process.env.CI
+    headless: process.env.CI ? true : false,
+    testIdAttribute: 'data-testid'
```
Comment (steven-the-qa) [inline playwright.config.ts:48]: This is the default in Playwright, but I want to make it explicit for transparency's sake.

File: playwright.config.ts [hunk 5]
```diff
@@ -24,88 +24,56 @@ const config: PlaywrightTestConfig = {
...
-    //   name: 'Google Chrome',
-    //   use: {
-    //     channel: 'chrome',
-    //   },
-    // },
   ],
 
   /* Folder for test artifacts such as screenshots, videos, traces, etc. */
-  // outputDir: 'test-results/',
+  outputDir: 'playwright-artifacts',
```
Comment (steven-the-qa) [inline playwright.config.ts:82]: This name is more accurate for what actually goes into this folder.

File: package.json [hunk 6]
```diff
@@ -22,7 +22,7 @@
     "vue-router": "^4.2.4"
   },
   "devDependencies": {
-    "@playwright/test": "^1.38.0",
+    "@playwright/test": "^1.39.0",
```
Comment (steven-the-qa) [inline package.json:25]: Latest release has some useful fixture & custom matcher features.

**Review by elliezub — APPROVED** [review]
Looks great Steven, also still works on my end! Glad you got it figured out 😄

**Pull request merged**